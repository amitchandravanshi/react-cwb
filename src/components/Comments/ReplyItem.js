import React from "react";
import { withStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import moment from "moment";
import CommentTarget from "./CommentTarget";
import CommentHeader from "./CommentHeader";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";

const styles = theme => ({
  card: {
    width: "86%",
    background: "#f3f4f6",
    margin: "1rem 0"
  },
  wrapper: {
    display: "flex",
    width: "92%",
    flexDirection: "row",
    margin: "auto",
    flexWrap: "wrap"
  },
  cardContent: {
    padding: 0
  },
  spacing: {
    width: "12%"
  },
  button: {
    color: "#487dde",
    textTransform: "lowercase"
  },
  fabButton: {
    margin: theme.spacing.unit,
    color: "#424242",
    backgroundColor: "#add8e6",
    height: "32px",
    minWidth: "10%",
    width: "12%",
    overflow: "hidden"
  }
});

const ReplyItem = props => {
  const { classes, comment } = props;
  let {
    updatedAt,
    commentId,
    userInitials,
    commentStatus,
    createdAt,
    commentValue,
    replyToId
  } = comment;
  return (
    <Card className={classes.card}>
      <CardContent className={classes.cardContent}>
        <div className={classes.wrapper}>
          <Button variant="fab" size="small" className={classes.fabButton}>
            {userInitials.substr(0, 2)}
          </Button>
          <CommentHeader
            title={userInitials}
            subheader={moment(createdAt)
              .startOf("second")
              .fromNow()}
            styles={{ width: "80%" }}
          />
          <div className={classes.spacing} />
          <CommentTarget targetText={commentValue} />
          {/* <div className={classes.spacing} />
          <Button className={classes.button}>Delete</Button> */}
        </div>
      </CardContent>
    </Card>
  );
};

export default withStyles(styles)(ReplyItem);
