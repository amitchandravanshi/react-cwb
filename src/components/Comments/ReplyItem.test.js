import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import ReplyItem from "./ReplyItem";
const props = {
  classes: {},
  comment: {
    updatedAt: new Date(),
    commentId: "12",
    userInitials: "AK",
    commentStatus: "processing",
    createdAt: new Date(),
    commentValue: "testing",
    replyToId: "234"
  }
};
describe("<CommentControls> ", () => {
  it("<CommentControls> snapshot ", () => {
    const wrapper = mount(<ReplyItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
