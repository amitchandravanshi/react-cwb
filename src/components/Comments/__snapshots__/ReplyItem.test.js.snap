// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentControls>  <CommentControls> snapshot  1`] = `
<ReplyItem
  classes={
    Object {
      "button": "ReplyItem-button-5",
      "card": "ReplyItem-card-1",
      "cardContent": "ReplyItem-cardContent-3",
      "fabButton": "ReplyItem-fabButton-6",
      "spacing": "ReplyItem-spacing-4",
      "wrapper": "ReplyItem-wrapper-2",
    }
  }
/>
`;
